libminifi/include/utils/FlatMap.h

```diff
@@ -282,20 +290,18 @@ class FlatMap{
     return data_.max_size();
   }
 
-  bool empty() const noexcept {
+  [[nodiscard]] bool empty() const noexcept {
     return data_.empty();
   }
 
-  bool contains(const K& key) const {
+  template<typename T>
+  requires std::equality_comparable_with<K, T>
+  bool contains(const T& key) const {
```

Comment:
```suggestion
  bool contains(std::equality_comparable_with<K> auto const& key) const {
```

Reply:
good idea, changed it and the others (except for operator[]) in https://github.com/apache/nifi-minifi-cpp/pull/1612/commits/f2092883c378f21a8d7ec2ab53c3826bcbef07c2

Reply:
it looks like MSVC doesnt like constraint auto, Im getting
```
C:\nifi-minifi-cpp\libminifi\test\unit\FlatMapTests.cpp(51): fatal error C1001: Internal compiler error.
(compiler file 'D:\a\_work\1\s\src\vctools\Compiler\CxxFE\sl\p1\c\trees.c', line 24533)
 To work around this problem, try simplifying or changing the program near the locations listed above.
If possible please provide a repro here: https://developercommunity.visualstudio.com/
```
I've found some reported issues with constraint auto and MSVC but those issues should be solved, ours is a bit more complex. I will try to create an MRE about this issue and file a bug with microsoft, in the meantime should I revert to explicit templates?

Reply:
I'd use the shortest for that's working with all targeted compilers. 
That translates to first fallback to this:
```c++
  template<std::equality_comparable_with<K> T>
  bool contains(T const& key) const {
```
and second fallback to the original version.

On an unrelated note, we should add `noexcept` wherever it makes sense. It may make our life easier down the line.

Reply:
Good idea, the first fallback works on MSVC aswell. https://github.com/apache/nifi-minifi-cpp/pull/1612/commits/617fc38829b51a666b87ab11bd68ff9daea16523
Adding noexcept where applicable is a good idea, I didnt find any obvious target on a quick glance, and I didnt want to burden this PR with a more complex analysis. Should I file a new jira for this?

Reply:
For `noexcept`, I was thinking `operator[]`, `find`, `erase` and `contains`. It may need to be conditional, e.g. `erase` does `swap`, so it needs to be `std::is_nothrow_swappable_v`. I was hoping for `noexcept` `find`, but it would have to rely on a complex condition, so it may not be feasible after all.

If nothing obvious stands out, don't bother with the new Jira.